lib/WP_Auth0_UsersRepo.php

```diff
@@ -68,90 +68,83 @@ public function tokenHasRequiredScopes( $jwt ) {
 
 	}
 
-	public function create( $userinfo, $token, $access_token = null, $role = null, $ignore_unverified_email = false ) {
-
-		// If the user doesn't exist we need to either create a new one, or assign them to an existing one
-		$isDatabaseUser = false;
-
-		if ( isset( $userinfo->identities ) ) {
+	/**
+	 * Create or join a WP user with an incoming Auth0 one or reject with an exception.
+	 *
+	 * @param object      $userinfo - Profile object from Auth0.
+	 * @param string      $token - ID token from Auth0.
+	 * @param null|string $access_token - TODO: Deprecate, not used
+	 * @param null|string $role - TODO: Deprecate, not used
+	 * @param bool        $skip_email_verified - TODO: Deprecate, not used
+	 *
+	 * @return int|null|WP_Error
+	 *
+	 * @throws WP_Auth0_CouldNotCreateUserException
+	 * @throws WP_Auth0_EmailNotVerifiedException
+	 * @throws WP_Auth0_RegistrationNotEnabledException
+	 */
+	public function create( $userinfo, $token, $access_token = null, $role = null, $skip_email_verified = false ) {
```

Comment:
This is the meat of this PR. This method was confusing to read and debug so just patching in the skip strategy logic would have probably taken as much time as a line-by-line re-write.